Fix unexpected smart pointer static analysis warnings in Source/WebKit/GPUProcess

rniwa opened the pull request:
#### cd38a1d1c8c83c9ad9e8fb0b7d40ac4c51cb2b06
<pre>
Fix unexpected smart pointer static analysis warnings in Source/WebKit/GPUProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=279346">https://bugs.webkit.org/show_bug.cgi?id=279346</a>

Reviewed by Chris Dumez and Mike Wyrzykowski.

Deploy more smart pointers in Source/WebKit/GPUProcess to fix the static analysis warnings.

* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
(WebKit::RemoteXRBinding::RemoteXRBinding):
(WebKit::RemoteXRBinding::protectedStreamConnection):
(WebKit::RemoteXRBinding::protectedBacking):
(WebKit::RemoteXRBinding::protectedGPU):
(WebKit::RemoteXRBinding::createProjectionLayer):
(WebKit::RemoteXRBinding::getViewSubImage):
(WebKit::RemoteXRBinding::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRProjectionLayer.cpp:
(WebKit::RemoteXRProjectionLayer::RemoteXRProjectionLayer):
(WebKit::RemoteXRProjectionLayer::protectedStreamConnection):
(WebKit::RemoteXRProjectionLayer::protectedGPU const):
(WebKit::RemoteXRProjectionLayer::connection const):
(WebKit::RemoteXRProjectionLayer::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRProjectionLayer.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRSubImage.cpp:
(WebKit::RemoteXRSubImage::RemoteXRSubImage):
(WebKit::RemoteXRSubImage::protectedStreamConnection):
(WebKit::RemoteXRSubImage::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRSubImage.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRView.cpp:
(WebKit::RemoteXRView::RemoteXRView):
(WebKit::RemoteXRView::protectedStreamConnection):
(WebKit::RemoteXRView::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRView.h:
* Source/WebKit/GPUProcess/media/RemoteLegacyCDMFactoryProxy.cpp:
(WebKit::RemoteLegacyCDMFactoryProxy::removeSession):
(WebKit::RemoteLegacyCDMFactoryProxy::logger const):

Canonical link: <a href="https://commits.webkit.org/283359@main">https://commits.webkit.org/283359@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/923e2f30b330526e9630bfe55181321be1b9bb72

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66055 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45428 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18674 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70086 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16664 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53227 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16946 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53021 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11604 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69122 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41914 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57187 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33655 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38585 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15541 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60479 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14911 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71789 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10010 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14323 "Found 1 new test failure: http/wpt/mediastream/mediastreamtrackprocessor-videoframe-timestamp.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60336 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10042 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57249 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60627 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8270 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1909 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9997 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41236 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42312 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43495 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42056 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
```diff
@@ -65,37 +80,39 @@ void RemoteXRBinding::createProjectionLayer(WebCore::WebGPU::TextureFormat color
         .scaleFactor = scaleFactor
     };
-    auto projectionLayer = m_backing->createProjectionLayer(WTFMove(init));
+    auto projectionLayer = protectedBacking()->createProjectionLayer(WTFMove(init));
```
`auto` -> `RefPtr` ?

mwyrzykowski replied:
Just wondering why is the call to `protectedBacking()` needed?

`m_backing` is already a `Ref`:

```
Ref<WebCore::WebGPU::XRBinding> m_backing;
```

rniwa replied:
Because m_backing can be cleared.

rniwa replied:
Oh oops, missed these review comments. Will fix.

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
```diff
@@ -65,37 +80,39 @@ void RemoteXRBinding::createProjectionLayer(WebCore::WebGPU::TextureFormat color
-    auto remoteProjectionLayer = RemoteXRProjectionLayer::create(*projectionLayer, m_objectHeap, m_streamConnection.copyRef(), m_gpu, identifier);
-    m_objectHeap->addObject(identifier, remoteProjectionLayer);
+    Ref objectHeap = m_objectHeap.get();
+    auto remoteProjectionLayer = RemoteXRProjectionLayer::create(*projectionLayer, objectHeap, protectedStreamConnection(), protectedGPU(), identifier);
```
`auto` -> `Ref`?

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
```diff
@@ -65,37 +80,39 @@ void RemoteXRBinding::createProjectionLayer(WebCore::WebGPU::TextureFormat color
     }
 
-    auto subImage = m_backing->getViewSubImage(*projectionLayer, eye);
+    auto subImage = protectedBacking()->getViewSubImage(*projectionLayer, eye);
```
`auto` -> `RefPtr`

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
```diff
@@ -65,37 +80,39 @@ void RemoteXRBinding::createProjectionLayer(WebCore::WebGPU::TextureFormat color
 
-    auto remoteSubImage = RemoteXRSubImage::create(*subImage, m_objectHeap, m_streamConnection.copyRef(), m_gpu, identifier);
-    m_objectHeap->addObject(identifier, remoteSubImage);
+    auto remoteSubImage = RemoteXRSubImage::create(*subImage, objectHeap, protectedStreamConnection(), protectedGPU(), identifier);
```
`auto` -> `Ref`

rniwa replied:
Fixed.

cdumez commented on Source/WebKit/GPUProcess/media/RemoteLegacyCDMFactoryProxy.cpp:
```diff
@@ -190,9 +190,10 @@ bool RemoteLegacyCDMFactoryProxy::allowsExitUnderMemoryPressure() const
+        m_logger = logger.ptr();
         auto connection = m_gpuConnectionToWebProcess.get();
-        m_logger->setEnabled(this, connection ? connection->sessionID().isAlwaysOnLoggingAllowed() : false);
+        logger->setEnabled(this, connection ? connection->sessionID().isAlwaysOnLoggingAllowed() : false);
```
`connection ? connection->sessionID().isAlwaysOnLoggingAllowed() : false` could be `connection && connection->sessionID().isAlwaysOnLoggingAllowed()`

rniwa replied:
Fixed.

cdumez reviewed (approved):
LGTM with nits.

rniwa committed:
Fix unexpected smart pointer static analysis warnings in Source/WebKit/GPUProcess
https://bugs.webkit.org/show_bug.cgi?id=279346

Reviewed by Chris Dumez and Mike Wyrzykowski.

Deploy more smart pointers in Source/WebKit/GPUProcess to fix the static analysis warnings.

* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.cpp:
(WebKit::RemoteXRBinding::RemoteXRBinding):
(WebKit::RemoteXRBinding::protectedStreamConnection):
(WebKit::RemoteXRBinding::protectedBacking):
(WebKit::RemoteXRBinding::protectedGPU):
(WebKit::RemoteXRBinding::createProjectionLayer):
(WebKit::RemoteXRBinding::getViewSubImage):
(WebKit::RemoteXRBinding::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRBinding.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRProjectionLayer.cpp:
(WebKit::RemoteXRProjectionLayer::RemoteXRProjectionLayer):
(WebKit::RemoteXRProjectionLayer::protectedStreamConnection):
(WebKit::RemoteXRProjectionLayer::protectedGPU const):
(WebKit::RemoteXRProjectionLayer::connection const):
(WebKit::RemoteXRProjectionLayer::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRProjectionLayer.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRSubImage.cpp:
(WebKit::RemoteXRSubImage::RemoteXRSubImage):
(WebKit::RemoteXRSubImage::protectedStreamConnection):
(WebKit::RemoteXRSubImage::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRSubImage.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRView.cpp:
(WebKit::RemoteXRView::RemoteXRView):
(WebKit::RemoteXRView::protectedStreamConnection):
(WebKit::RemoteXRView::stopListeningForIPC):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteXRView.h:
* Source/WebKit/GPUProcess/media/RemoteLegacyCDMFactoryProxy.cpp:
(WebKit::RemoteLegacyCDMFactoryProxy::removeSession):
(WebKit::RemoteLegacyCDMFactoryProxy::logger const):

Canonical link: https://commits.webkit.org/283359@main
